### ISSUE TITLE
Values cannot be replaced, only appended to or removed entirely

### ISSUE THREAD

**Issue by andrewmyhre** [kubernetes/cloud-provider-aws#428]
Due to [this unintuitive behavior in Helm](https://github.com/helm/helm/issues/6465), it is not possible to override the nodeSelector or tolerations with just values specified in an override values.yaml. This is challenge because in recent versions of Kubernetes the standard label for control-plane nodes has changed from `node-role.kubernetes.io/master` to `node-role.kubernetes.io/control-plane`, and it is not possible to specify overrides for the `tolerations` and `nodeSelector` values which *remove* `node-role.kubernetes.io/master`. Because this is not considered a bug by Helm and will not be changed, and only way to install this chart on a Kubernetes v1.24 cluster is to fork it and remove the `nodeSelector` and `tolerations` default values.

**What happened**: 

Create a values file with the following:
```
nodeSelector:
  node-role.kubernetes.io/control-plane: ""

tolerations:
- key: node.cloudprovider.kubernetes.io/uninitialized
  value: "true"
  effect: NoSchedule
- key: node-role.kubernetes.io/control-plane
  effect: NoSchedule

```

Using these values, the final daemonset is rendered as:
```
apiVersion: apps/v1
kind: DaemonSet
metadata:
  name: aws-cloud-controller-manager
  labels:
    k8s-app:  aws-cloud-controller-manager
    helm.sh/chart: "aws-cloud-controller-manager-0.0.6"
  namespace: kube-system
spec:
  selector:
    matchLabels:
      k8s-app:  aws-cloud-controller-manager
  updateStrategy:
    type: RollingUpdate
  template:
    metadata:
      name: aws-cloud-controller-manager
      labels:
        k8s-app:  aws-cloud-controller-manager
    spec:
      tolerations:
        - effect: NoSchedule
          key: node.cloudprovider.kubernetes.io/uninitialized
          value: "true"
        - effect: NoSchedule
          key: node-role.kubernetes.io/master
        - effect: NoSchedule
          key: node-role.kubernetes.io/control-plane
      nodeSelector:
        node-role.kubernetes.io/control-plane: ""
        node-role.kubernetes.io/master: ""
```

**What you expected to happen**:
I expected the rendered output to match the values I specified exactly.

**How to reproduce it (as minimally and precisely as possible)**:
(Start with a v1.24+ Kubernetes Kubeadm provisioned cluster)

1. Create a values file with the following content:
```
nodeSelector:
  node-role.kubernetes.io/control-plane: ""

tolerations:
- key: node.cloudprovider.kubernetes.io/uninitialized
  value: "true"
  effect: NoSchedule
- key: node-role.kubernetes.io/control-plane
  effect: NoSchedule

```
2. `helm template aws-cloud-controller-manager/aws-cloud-controller-manager --values <your values file>`
3. The daemonset will not be able to schedule any pods.

**Anything else we need to know?**:

**Environment**:
- Kubernetes version (use `kubectl version`):
```
Client Version: version.Info{Major:"1", Minor:"24", GitVersion:"v1.24.2", GitCommit:"f66044f4361b9f1f96f0053dd46cb7dce5e990a8", GitTreeState:"clean", BuildDate:"2022-06-15T14:22:29Z", GoVersion:"go1.18.3", Compiler:"gc", Platform:"linux/amd64"}
Kustomize Version: v4.5.4
Server Version: version.Info{Major:"1", Minor:"24", GitVersion:"v1.24.2", GitCommit:"f66044f4361b9f1f96f0053dd46cb7dce5e990a8", GitTreeState:"clean", BuildDate:"2022-06-15T14:15:38Z", GoVersion:"go1.18.3", Compiler:"gc", Platform:"linux/amd64"}
```
- Cloud provider or hardware configuration: AWS EC2
- OS (e.g. from /etc/os-release): Ubuntu 2004
- Kernel (e.g. `uname -a`): Linux
- Install tools: kubeadm
- Others:

/kind bug

**Comment by kwohlfahrt** [comment]
FWIW, I was able to schedule the controller manager using the following config:
```yaml
image:
  tag: v1.24.1
nodeSelector:
  node-role.kubernetes.io/master: null
  node-role.kubernetes.io/control-plane: ""
tolerations:
  - key: node-role.kubernetes.io/control-plane
    effect: NoSchedule
  - key: node-role.kubernetes.io/master
    effect: NoSchedule
args:
  - --cloud-provider=aws
  - --cluster-cidr=$CLUSTER_CIDR
  - --cluster-name=$CLUSTER_NAME
```

Note the `nodeSelector` has to use a `null` value to remove the old `master` selector (as per helm/helm#1966), but both taints have to be tolerated, because they are both still present until `v1.25`.

This schedules one instance of the pod, which is correct (this is a single-master test):
```
# kubectl -n kube-system get daemonset aws-cloud-controller-manager
NAME                           DESIRED   CURRENT   READY   UP-TO-DATE   AVAILABLE   NODE SELECTOR                            AGE
aws-cloud-controller-manager   1         1         0       1            0           node-role.kubernetes.io/control-plane=   89s
```

**Comment by andrewmyhre** [comment]
Ah thanks for the tip

**Comment by andrewmyhre** [comment]
@kwohlfahrt Finally got around to testing this, it actually doesn't work as you described. Here's what's produced when I try this:

```
Error: unable to build kubernetes objects from release manifest: error validating "": error validating data: unknown object type "nil" in DaemonSet.spec.template.spec.nodeSelector.node-role.kubernetes.io/master
```

**Comment by andrewmyhre** [comment]
@kwohlfahrt I see the above behavior with 0.0.7 but not with 0.0.6

**Comment by kwohlfahrt** [comment]
I think this specific case may have been fixed in #466, so it make sense that using `null` to delete a value that isn't present might not work? I don't have the time to test this again at the moment.
